ports/raspberrypi/common-hal/busio/UART.c

```diff
@@ -24,238 +24,135 @@
-//     // CHSIZE is 0 for 8 bits, 5, 6, 7 for 5, 6, 7 bits. 1 for 9 bits, but we don't support that.
-//     sercom->USART.CTRLB.reg &= ~(SERCOM_USART_CTRLB_TXEN |
-//                                  SERCOM_USART_CTRLB_RXEN |
-//                                  SERCOM_USART_CTRLB_PMODE |
-//                                  SERCOM_USART_CTRLB_SBMODE |
-//                                  SERCOM_USART_CTRLB_CHSIZE_Msk);
-//     sercom->USART.CTRLB.reg |= (have_tx ? SERCOM_USART_CTRLB_TXEN : 0) |
-//                                (have_rx ? SERCOM_USART_CTRLB_RXEN : 0) |
-//                                (parity == BUSIO_UART_PARITY_ODD ? SERCOM_USART_CTRLB_PMODE : 0) |
-//                                (stop > 1 ? SERCOM_USART_CTRLB_SBMODE : 0) |
-//                                SERCOM_USART_CTRLB_CHSIZE(bits % 8);
-
-//     // Set baud rate
-//     common_hal_busio_uart_set_baudrate(self, baudrate);
-
-//     // Turn on rx interrupt handling. The UART async driver has its own set of internal callbacks,
-//     // which are set up by uart_async_init(). These in turn can call user-specified callbacks.
-//     // In fact, the actual interrupts are not enabled unless we set up a user-specified callback.
-//     // This is confusing. It's explained in the Atmel START User Guide -> Implementation Description ->
-//     // Different read function behavior in some asynchronous drivers. As of this writing:
-//     // http://start.atmel.com/static/help/index.html?GUID-79201A5A-226F-4FBB-B0B8-AB0BE0554836
-//     // Look at the ASFv4 code example for async USART.
-//     usart_async_register_callback(usart_desc_p, USART_ASYNC_RXC_CB, usart_async_rxc_callback);
-
-
-//     if (have_tx) {
-//         gpio_set_pin_direction(tx->number, GPIO_DIRECTION_OUT);
-//         gpio_set_pin_pull_mode(tx->number, GPIO_PULL_OFF);
-//         gpio_set_pin_function(tx->number, tx_pinmux);
-//         self->tx_pin  = tx->number;
-//         claim_pin(tx);
-//     } else {
-//         self->tx_pin = NO_PIN;
-//     }
-
-//     if (have_rx) {
-//         gpio_set_pin_direction(rx->number, GPIO_DIRECTION_IN);
-//         gpio_set_pin_pull_mode(rx->number, GPIO_PULL_OFF);
-//         gpio_set_pin_function(rx->number, rx_pinmux);
-//         self->rx_pin  = rx->number;
-//         claim_pin(rx);
-//     } else {
-//         self->rx_pin = NO_PIN;
-//     }
-
-//     usart_async_enable(usart_desc_p);
+        const mcu_pin_obj_t * tx, const mcu_pin_obj_t * rx,
+        const mcu_pin_obj_t * rts, const mcu_pin_obj_t * cts,
+        const mcu_pin_obj_t * rs485_dir, bool rs485_invert,
+        uint32_t baudrate, uint8_t bits, busio_uart_parity_t parity, uint8_t stop,
+        mp_float_t timeout, uint16_t receiver_buffer_size, byte* receiver_buffer,
+        bool sigint_enabled) {
+
+    if ((rs485_dir != NULL) || (rs485_invert)) {
+        mp_raise_NotImplementedError(translate("RS485 is not supported on this board"));
```

Comment:
Is it possible to implement? If so, then lets match the existing error "RS485 Not yet supported on this device"

Reply:
It might be... I'll change it to _not yet_ for now.